app/bundles/PageBundle/EventListener/BuildJsSubscriber.php

```diff
@@ -396,7 +396,11 @@ public function onBuildJsForVideo(BuildJsEvent $event)
     
     MauticJS.iterateCollection(videoElements)(function(node, i){
         var playerFeatures = [];
-        var source   = node.getElementsByTagName('source')[0];
+        var sourceNodes = node.getElementsByTagName('source');
+        if (sourceNodes.length === 0) {
```

Comment:
by returning here if there is no source node the functionality will still not work, even if the source is specified in the src attribute. I suggest to execute the following couple of lines only if sourceNodes.length > 0 and execute the rest of the function even if sourceNodes.length === 0. That way the functionality will also work for video tags with the src attribute